kornia/morphology/morphology.py

```diff
@@ -44,27 +61,54 @@ def dilation(tensor: torch.Tensor, kernel: torch.Tensor, origin: Optional[List[i
 
     # pad
     pad_e: List[int] = [origin[1], se_w - origin[1] - 1, origin[0], se_h - origin[0] - 1]
-    output: torch.Tensor = F.pad(tensor, pad_e, mode='constant', value=0.0)
+    if border_type == 'geodesic':
+        border_value = -infinity
+        border_type = 'constant'
+    output: torch.Tensor = F.pad(tensor, pad_e, mode=border_type, value=border_value)
 
     # computation
+    if structuring_element is None:
+        neighborhood = torch.zeros_like(kernel)
+        neighborhood[kernel == 0] = -infinity
+    else:
+        neighborhood = structuring_element
+        neighborhood[kernel == 0] = -infinity
+
     output = output.unfold(2, se_h, 1).unfold(3, se_w, 1)
-    output, _ = torch.max(output + kernel.flip((0, 1)), 4)
+    output, _ = torch.max(output + neighborhood.flip((0, 1)), 4)
     output, _ = torch.max(output, 4)
 
     return output
 
 
 # Erosion
-def erosion(tensor: torch.Tensor, kernel: torch.Tensor, origin: Optional[List[int]] = None) -> torch.Tensor:
+def erosion(
+    tensor: torch.Tensor,
+    kernel: torch.Tensor,
+    structuring_element: Optional[torch.Tensor] = None,
+    origin: Optional[List[int]] = None,
+    border_type: str = 'geodesic',
+    border_value: float = 0.0,
+    infinity: float = 1e4,
```

Comment:
```suggestion
    eps: float = 1e4,
```
then in the description mention that is an infinity value. Same for the others